### PR TITLE
[refactor] readme: add hosting instructions for bringing own reverse …

### LINKED ISSUE #20: Changing server port cannot be done via settings.yaml
Setting
```yaml
server:
  port: 1234
```
in `settings.yaml` does not make the Docker container listen on 0.0.0.0:1234.

Only setting
```yaml
    environment:
      BIND_ADDRESS: 0.0.0.0:1234
```
in `docker-compose.yaml` makes it work.

`settings.yaml` is not taken into account for the port, but there is a workaround.
Does setting the `settings.yaml` port number do anything?

EDIT: I found the workaround by looking at the entrypoint script of the Dockerfile https://github.com/searxng/searxng/blob/31521f3a465fa2beb7e3fd7ed47621262a28f554/dockerfiles/docker-entrypoint.sh#L26-L27:
```sh
export DEFAULT_BIND_ADDRESS="0.0.0.0:8080"
export BIND_ADDRESS="${BIND_ADDRESS:-${DEFAULT_BIND_ADDRESS}}"
```

### PR DESCRIPTION
**Pull request by Bnyro**
…proxy

closes #20

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,17 +22,12 @@ Create a new SearXNG instance in five minutes using Docker
   ```
 - Edit the [.env](https://github.com/searxng/searxng-docker/blob/master/.env) file to set the hostname and an email
 - Generate the secret key `sed -i "s|ultrasecretkey|$(openssl rand -hex 32)|g" searxng/settings.yml`
-- Edit the [searxng/settings.yml](https://github.com/searxng/searxng-docker/blob/master/searxng/settings.yml) file according to your need
-- Check everything is working: `docker compose up`
-- Run SearXNG in the background: `docker compose up -d`
+- Edit [searxng/settings.yml](https://github.com/searxng/searxng-docker/blob/master/searxng/settings.yml) according to your needs
```
Comment (return42) [inline README.md:25]: > The first few steps are the same for both installation methods however.

Aren't these steps not needed for both installation methods?

To be more clear what is the *first*, *second*, *third*, .. step we may better use a numbered list? ..

1. [Install docker](https://docs.docker.com/install/)
2. Get searxng-docker
3. Generate the secret key `sed -i "s|ultrasecretkey|$(openssl rand -hex 32)|g" searxng/settings.yml`
4. Edit the [.env](https://github.com/searxng/searxng-docker/blob/master/.env) file to set the hostname and an email
5. Edit [searxng/settings.yml](https://github.com/searxng/searxng-docker/blob/master/searxng/settings.yml) according to your needs

Comment (Bnyro) [reply README.md]: Yeah, good idea :+1:

**Review by return42 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -40,8 +36,16 @@ Create a new SearXNG instance in five minutes using Docker
...
-> On the first run, you must remove `cap_drop: - ALL` from the `docker-compose.yaml` file for the `searxng` service to successfully create `/etc/searxng/uwsgi.ini`. This is necessary because the `cap_drop: - ALL` directive removes all capabilities, including those required for the creation of the `uwsgi.ini` file.
+### Method 1: With Caddy included (recommended for beginners)
+6. Check everything is working: `docker compose up`
+7. Run SearXNG in the background: `docker compose up -d`
+
+### Method 2: Bring your own reverse proxy (experienced users)
+6. Remove the caddy related parts in `docker-compose.yaml` such as the caddy service and its volumes.
+7. Point your reverse proxy to the port set for the `searxng` service in `docker-compose.yml` (8080 by default). You can change the port `searxng` listens on inside the docker container (e.g. if you want to operate in `host` network mode) with the `BIND_ADDRESS` environment variable (defaults to `0.0.0.0:8080`). You can just set the environment variable inside `docker-compose.yaml`.
+8. Generate and configure the required TLS certificates with the reverse proxy of your choice.
+9. Check everything is working: `docker compose up`
```
Comment (unixfox) [inline README.md:47]: Wouldn't recommend that. One can always do up -d then logs -f to check if it works.

**Review by unixfox — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -40,8 +36,16 @@ Create a new SearXNG instance in five minutes using Docker
...
 
-> [!NOTE]  
-> On the first run, you must remove `cap_drop: - ALL` from the `docker-compose.yaml` file for the `searxng` service to successfully create `/etc/searxng/uwsgi.ini`. This is necessary because the `cap_drop: - ALL` directive removes all capabilities, including those required for the creation of the `uwsgi.ini` file.
+### Method 1: With Caddy included (recommended for beginners)
+6. Check everything is working: `docker compose up`
+7. Run SearXNG in the background: `docker compose up -d`
+
+### Method 2: Bring your own reverse proxy (experienced users)
+6. Remove the caddy related parts in `docker-compose.yaml` such as the caddy service and its volumes.
+7. Point your reverse proxy to the port set for the `searxng` service in `docker-compose.yml` (8080 by default). You can change the port `searxng` listens on inside the docker container (e.g. if you want to operate in `host` network mode) with the `BIND_ADDRESS` environment variable (defaults to `0.0.0.0:8080`). You can just set the environment variable inside `docker-compose.yaml`.
```
Comment (unixfox) [inline README.md:45]: There are too many things in one single point. Please try to split them. or say like:

- the solutions:
  - solution A
  - solution B

**Review by unixfox — CHANGES_REQUESTED** [review]

**Pull request merged**